src/main/java/org/springframework/integration/aws/config/xml/SqsOutboundChannelAdapterParser.java

```diff
@@ -34,13 +34,37 @@
  */
 public class SqsOutboundChannelAdapterParser extends AbstractOutboundChannelAdapterParser {
 
+    public static final String QUEUE_MESSAGING_TEMPLATE_REF = "queue-messaging-template";
+
+    private boolean hasAttribute(Element element, String attribute) {
+        return StringUtils.hasText(element.getAttribute(attribute));
+    }
+
 	@Override
 	protected AbstractBeanDefinition parseConsumer(Element element, ParserContext parserContext) {
 		BeanDefinitionBuilder builder = BeanDefinitionBuilder.genericBeanDefinition(SqsMessageHandler.class);
-		builder.addConstructorArgReference(element.getAttribute(AmazonWSParserUtils.SQS_REF));
-		if (StringUtils.hasText(element.getAttribute(AmazonWSParserUtils.RESOURCE_ID_RESOLVER_REF))) {
+
+        if (hasAttribute(element, QUEUE_MESSAGING_TEMPLATE_REF) &&
+                (
+                        hasAttribute(element, AmazonWSParserUtils.SQS_REF) ||
+                        hasAttribute(element, AmazonWSParserUtils.RESOURCE_ID_RESOLVER_REF))) {
+            parserContext.getReaderContext().error(QUEUE_MESSAGING_TEMPLATE_REF + " should not be defined in conjunction with " + AmazonWSParserUtils.SQS_REF + " or " + AmazonWSParserUtils.RESOURCE_ID_RESOLVER_REF, element);
+
+        }
+
+        if (!hasAttribute(element, QUEUE_MESSAGING_TEMPLATE_REF) && !hasAttribute(element, AmazonWSParserUtils.SQS_REF)) {
+            parserContext.getReaderContext().error("One of " + QUEUE_MESSAGING_TEMPLATE_REF + " or " + AmazonWSParserUtils.SQS_REF + " must be defined.", element);
+        }
+
+        if (hasAttribute(element, AmazonWSParserUtils.SQS_REF)) {
+            builder.addConstructorArgReference(element.getAttribute(AmazonWSParserUtils.SQS_REF));
+        }
+		if (hasAttribute(element, AmazonWSParserUtils.RESOURCE_ID_RESOLVER_REF)) {
 			builder.addConstructorArgReference(element.getAttribute(AmazonWSParserUtils.RESOURCE_ID_RESOLVER_REF));
 		}
+        if (hasAttribute(element, QUEUE_MESSAGING_TEMPLATE_REF)) {
+            builder.addConstructorArgReference(element.getAttribute(QUEUE_MESSAGING_TEMPLATE_REF));
```

Comment:
The code above looks good and we definitely can merge it, but for the future would be better to extract local vars to avoid duplicate method invocation for those attr. Plus variables like `hasSqs` for the `StringUtils.hasText(element.getAttribute(AmazonWSParserUtils.SQS_REF))`